openwisp_monitoring/device/admin.py

```diff
@@ -275,6 +275,22 @@ def health_status(self, obj):
 
     health_status.short_description = _('health status')
 
+    def change_view(self, request, object_id, form_url='', extra_context=None):
+        device = self.get_object(request, object_id)
```

Comment:
This will add a duplicate query to the view generation.
See how this query is repeated later on:
https://github.com/django/django/blob/ef62a3a68c9d558486145a42c0d71ea9a76add9e/django/contrib/admin/options.py#L1769

It should be possible to avoid this.
There's several ways. What about overriding get_object instead? That way we would also avoid other issues.
For example, with this solution I am pretty sure that if you try to access a device with a valid UUID but which does not exist an uncaught exception will be raised.